tmail-backend/integration-tests/jmap/jmap-integration-tests-common/src/main/scala/com/linagora/tmail/james/common/TeamMailboxMemberSetMethodContract.scala

```diff
@@ -0,0 +1,864 @@
+           |  "methodCalls": [
+           |    [
+           |      "TeamMailboxMember/set",
+           |      {
+           |        "accountId": "29883977c13473ae7cb7678ef767cbfbaffc8a44a6e463d971d23a65c1dc4af6",
+           |        "update": {
+           |            "${teamMailbox.asString()}": {
+           |                "${CEDRIC.asString()}": {"role": "member"},
+           |                "abc@domain.tld": {"role": "member"}
+           |            }
+           |        }
+           |      },
+           |      "c1"
+           |    ]
+           |  ]
+           |}""".stripMargin)
+      .when
+      .post
+      .`then`
+      .statusCode(SC_OK)
+      .contentType(JSON)
+      .body("methodResponses[0]", jsonEquals(
+        s"""[
+           |    "TeamMailboxMember/set",
+           |    {
+           |        "accountId": "29883977c13473ae7cb7678ef767cbfbaffc8a44a6e463d971d23a65c1dc4af6",
+           |        "updated": {},
+           |        "notUpdated": {
+           |            "${teamMailbox.asString()}": {
+           |                "type": "invalidPatch",
+           |                "description": "Some users do not exist in the system"
+           |            }
+           |        }
+           |    },
+           |    "c1"
+           |]""".stripMargin).withOptions(ImmutableList.of(IGNORING_ARRAY_ORDER)))
+  }
+
+  @Test
+  def updateShouldReturnNotUpdatedWhenUserIsNotMemberOfTheTeamMailbox(server: GuiceJamesServer): Unit = {
+    val teamMailbox = TeamMailbox(DOMAIN, TeamMailboxName("hiring"))
+
+    server.getProbe(classOf[TeamMailboxProbe])
+      .create(teamMailbox)
+      .addManager(teamMailbox, ANDRE)
+
+    `given`
+      .body(
+        s"""{
+           |  "using": ["urn:ietf:params:jmap:core", "urn:ietf:params:jmap:mail", "com:linagora:params:jmap:team:mailboxes"],
+           |  "methodCalls": [
+           |    [
+           |      "TeamMailboxMember/set",
+           |      {
+           |        "accountId": "29883977c13473ae7cb7678ef767cbfbaffc8a44a6e463d971d23a65c1dc4af6",
+           |        "update": {
+           |            "${teamMailbox.asString()}": {
+           |                "${CEDRIC.asString()}": {"role": "member"}
+           |            }
+           |        }
+           |      },
+           |      "c1"
+           |    ]
+           |  ]
+           |}""".stripMargin)
+      .when
+      .post
+      .`then`
+      .statusCode(SC_OK)
+      .contentType(JSON)
+      .body("methodResponses[0]", jsonEquals(
+        s"""[
+           |    "TeamMailboxMember/set",
+           |    {
+           |        "accountId": "29883977c13473ae7cb7678ef767cbfbaffc8a44a6e463d971d23a65c1dc4af6",
+           |        "updated": {},
+           |        "notUpdated": {
+           |            "${teamMailbox.asString()}": {
+           |                "type": "invalidPatch",
+           |                "description": "Team mailbox is not found"
```

Comment:
Return error message `Team mailbox is not found` when a non-member user tries to update an existing team mailbox sounds misleading to me.

Maybe we could return something like "not manager of the team mailbox`?

Reply:
fixed